lib/install.js

```diff
@@ -581,7 +580,14 @@ Installer.prototype.saveToDependencies = function (cb) {
   validate('F', arguments)
   if (this.failing) return cb()
   log.silly('install', 'saveToDependencies')
-  saveRequested(this.args, this.idealTree, cb)
+  if (
+    this.differences.length ||
+    (this.idealTree.removedChildren || []).length
+  ) {
+    saveRequested(this.idealTree, cb)
```

Comment:
I think we need to save unconditionally, because:
1. I think this generates the initial `package-lock.json` even when you have an up-to-date `node_modules`.
2. We want to update the `packageIntegrity` even if it's something else that changed its hash